usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -7239,6 +7238,21 @@ CK_RV ep11tok_sign_init(STDLL_TokData_t * tokdata, SESSION * session,
         ctx->context = ep11_sign_state;
         ctx->context_len = ep11_sign_state_l;
         ctx->pkey_active = FALSE;
+        if (mech != &ctx->mech) { /* deferred init dup'ed mech already */
+            ctx->mech.mechanism = mech->mechanism;
+            if (mech->ulParameterLen > 0 && mech->pParameter != NULL) {
+                ctx->mech.pParameter = (CK_BYTE *) malloc(mech->ulParameterLen);
+                if (ctx->mech.pParameter == NULL) {
```

Comment:
What if ctx->mech.pParameter was not NULL before? Would we have to free the old buffer before allocating and assigning the new one?

Reply:
If we are in an init call at that point, it has already been checked that the context is not already active. In that case, ctx->mech.pParameter is supposed to be NULL.